packages/cosmic-swingset/lib/ag-solo/web.js

```diff
@@ -78,20 +78,23 @@ export async function makeHTTPListener(basedir, port, host, rawInboundCommand) {
       return true;
     }
 
-    // Validate the private accessToken.
-    const accessToken = await getAccessToken(port);
-    const reqToken = new URL(`http://localhost${req.url}`).searchParams.get(
-      'accessToken',
-    );
-
-    if (reqToken !== accessToken) {
-      log.error(
-        id,
-        `Invalid access token ${JSON.stringify(
-          reqToken,
-        )}; try running "agoric open"`,
+    // Bypass accessToken just for the wallet bridge.
+    if (req.url !== '/private/wallet-bridge') {
```

Comment:
It's not obvious to me why this URL should be exempted from the token check. I think this URL is used for a websocket, which might mean the token has to be delivered differently (inline, in a message, rather than as a query param?), but we still need access control on the websocket, right?

Reply:
The dapp is an unprivileged web page that needs to use the wallet bridge to post offers/etc.  The wallet bridge itself exposes a public facet with its own access control (managed within the wallet, which is protected by the access token).  If we need to improve security of the wallet bridge, it should be done in the context of that public facet rather than at the connection level.

It would make sense to give out an origin-specific access token that the dapp can store in local storage after the user approves the dapp's connection (and gives it a petname).  That would make it more resistant to other local users.  I'd like to postpone that to a future PR (slated during the alpha timeframe, but not tomorrow) if possible.

Reply:
Ok, does this capture the rule set? (if so, maybe worth adding as a comment):

* `path` outside `/private`: always accept
* `path` in `/private` but outside `/private/wallet-bridge`: require correct `accessToken=` in query params
* `path` in `/private/wallet-bridge`: origin-based access control: accept `chrome-extension:`, `moz-extension:`, and `http:`/`https:` `localhost`/`127.0.0.1`

Reply:
BTW, let's look for a pattern that makes this distinction between protected and public resources more explicit. Ideally we'd have two different ports entirely. I'm concerned about disagreement between the path prefixes that this code looks for (especially fencepost errors with a trailing slash) and the code elsewhere that accepts a request and relies upon us here to perform the auth check. "Don't separate designation from authority".

Reply:
I'm not concerned about fencepost errors.  The only URL that is relaxed is exactly `/private/wallet-bridge`, not `/private/wallet-bridge/`.  If they don't match that URL, then accessToken is required in addition to origin checking.